interp/numTok.ml

```diff
@@ -84,31 +91,31 @@ struct
     (* nasty code to remove when switching to zarith
        since zarith's of_string handles hexadecimal *)
     match UnsignedNat.classify n with
-    | CDec -> Bigint.of_string (to_string (sign,n))
+    | CDec -> Z.of_string (to_string (sign,n))
     | CHex ->
        let int_of_char c = match c with
          | 'a'..'f' -> 10 + int_of_char c - int_of_char 'a'
          | _ -> int_of_char c - int_of_char '0' in
-       let c16 = Bigint.of_int 16 in
+       let c16 = Z.of_int 16 in
        let s = UnsignedNat.to_string n in
-       let n = ref Bigint.zero in
+       let n = ref Z.zero in
        let len = String.length s in
        for d = 2 to len - 1 do
-         n := Bigint.(add (mult !n c16) (of_int (int_of_char s.[d])))
+         n := Z.(add (mul !n c16) (of_int (int_of_char s.[d])))
        done;
-       match sign with SPlus -> !n | SMinus -> Bigint.neg !n
+       match sign with SPlus -> !n | SMinus -> Z.neg !n
   let to_bigint n = bigint_of_string n
   let string_of_nonneg_bigint c n =
```

Comment:
```suggestion
  let string_of_nonneg_bigint c n =
    match c with
    | CDec -> Z.format "%d" n
    | CHex -> "0x" ^ Z.format "%x" n
```
(and delete all remaining of the function)

Reply:
Done, thanks!